whylog_vim/output_formater/teacher_formater/__init__.py

```diff
@@ -24,64 +24,72 @@ def format_parser(self, output, parser, parser_id, effect=True):
 
     def _format_line_info(self, output, parser, effect):
         output.add(
-            ParserOutputs.MESSAGE_CONTENT % (get_parser_name(parser._id), parser.line_content)
+            ParserOutputs.MESSAGE_CONTENT % (get_parser_name(parser.line_id), parser.line_content)
+        )
+        output.create_button(
+            partial(self.teacher_proxy.edit_line_content, parser), (
+                FunctionNames.EDIT_LINE_CONTENT, parser.line_id
+            )
         )
-        output.create_button(partial(self.teacher_proxy.edit_line_content, parser._id))
         output.add(ParserOutputs.META % (parser.line_resource_location, parser.line_offset))
         if not effect:
             output.add(ParserOutputs.COPY_BUTTON)
             output.add(ParserOutputs.DELETE_BUTTON)
-            output.create_button(partial(self.teacher_proxy.delete_parser, parser._id))
+            # output.create_button(partial(self.teacher_proxy.delete_parser, parser._id))
         output.add('')
 
     def _format_regexes(self, output, parser):
         output.add(ParserOutputs.REGEX_HEADER % parser.pattern_name)
         output.add(parser.pattern)
-        output.create_button(partial(self.teacher_proxy.edit_regex, parser._id))
+        output.create_button(
+            partial(self.teacher_proxy.edit_regex, parser), (
+                FunctionNames.EDIT_REGEX, parser.line_id
+            )
+        )
         output.add(ParserOutputs.GUESS_BUTTON)
-        output.create_button(partial(self.teacher_proxy.guess_regex, parser._id))
+        # output.create_button(partial(self.teacher_proxy.guess_regex, parser))
```

Comment:
In this PR I change lot of interface. If Paweł accept it I will remove this comment. Now this code is not compatible so I commented it and wait for clues from Paweł.